ArduPlane/tiltrotor.cpp

```diff
@@ -782,4 +782,27 @@ bool Tiltrotor::tilt_over_max_angle(void) const
     return (current_tilt > MIN(tilt_threshold, get_forward_flight_tilt()));
 }
 
+// throttle of forward flight motors including any tilting motors
+bool Tiltrotor::get_forward_throttle(float &throttle) const
+{
+    if (_is_vectored) {
+        float throttle_sum = 0.0f;
+        uint8_t num_vectored_motors = 0;
+        for (uint8_t i = 0; i < AP_MOTORS_MAX_NUM_MOTORS; ++i) {
+            if (is_motor_tilting(i)) {
+                float thr_out;
+                if (motors->get_raw_motor_throttle(i, thr_out)) {
```

Comment:
I think we can avoid the addition of the get_raw_motor_throttle() function by adding a variable that adds up the thrust on tilting motors in Tiltrotor::tilt_compensate_angle()

Reply:
I wanted the raw throttle value for each motor, not the thrust value because the throttle  value is equivalent to what the FW controller needs to demand to achieve the same PWM output whereas the the thrust value is related to PWM via a nonlinear function.

Reply:
We have the same issue for tailsitters there we apply the inverse of the linearisation. 

https://github.com/ArduPilot/ardupilot/blob/0ecdae05ba028a06cf939ad294f9d41d5df2dcb1/ArduPlane/tailsitter.cpp#L313-L314

The `actuator_to_thrust` and `thrust_to_actuator` functions allow you to go back and forth. However, you would still need to filter by tilting/not tilting.

Reply:
I've pushed a patch that uses the thrust_to_actuator function and reverted the patch that added the additional get_raw_motor_throttle function which should save on some flash space.